### PR TITLE
argv: move options_only to cli/args

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#5730

### CODE REVIEW

File: Library/Homebrew/cli/args.rb [hunk 1]
```diff
@@ -12,6 +12,23 @@ def initialize(argv:)
...
+        else
+          "--#{name.tr("_", "-").tr("?", "")}"
+        end
+      end
+
+      def options_only
+        self.to_h.keys
+          .map(&:to_s)
+          .reject { |name| name == "argv"  || name == "remaining"}
+          .map { |name| to_cli_option(name) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cli/args.rb]: Could do `.map(&method(:to_cli_option)`

**Comment by GauthamGoli** [comment]
@MikeMcQuaid Fixed styles.

**Comment by MikeMcQuaid** [comment]
Thanks again @GauthamGoli!

File: Library/Homebrew/cmd/cat.rb [hunk 2]
```diff
@@ -25,6 +25,6 @@ def cat
     raise "`brew cat` doesn't support multiple arguments" if args.remaining.size > 1
 
     cd HOMEBREW_REPOSITORY
-    safe_system "cat", formulae.first.path, *ARGV.options_only
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/cat.rb]: Meant to comment on this before: can the `ARGV.options_only` method be removed entirely now? If so and if it has no usage in Homebrew/homebrew-core: please remove it. Thanks!

Comment (GauthamGoli) [reply Library/Homebrew/cmd/cat.rb]: It has one usage in `formula_installer.rb`

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/cat.rb]: Cool. What's blocking that removal?

Comment (GauthamGoli) [reply Library/Homebrew/cmd/cat.rb]: `formula_installer.rb` and `build.rb` directly read and modify `ARGV` https://github.com/Homebrew/brew/issues/5730#issue-409874027

**Comment by GauthamGoli** [comment]
This implementation is buggy and we would revert it;

Lets say I pass `-v` switch, then `Homebrew.args` would have  `v?=true, verbose?=true`
and `Homebrew.args.options_only` would return both [`-v`, `--verbose`] which is wrong,
where as `ARGV.options_only` correctly returns `[-v]`

**Pull request merged**